src/vm/dwreport.cpp

```diff
@@ -2426,9 +2440,11 @@ FaultReportResult DoFaultReportWorker(      // Was Watson attempted, successful?
     pWatsonSharedMemory->bfmsoctdsLetRun = offerFlags;
 
     {
+        PathString wzModuleFileName;
         DWORD dwRet = WszGetModuleFileName(NULL,
-	                                       pWatsonSharedMemory->wzModuleFileName,
-	                                       NumItems(pWatsonSharedMemory->wzModuleFileName));
+	                                       wzModuleFileName);
+        BaseBucketParamsManager::CopyStringToBucket(pWatsonSharedMemory->wzModuleFileName, NumItems(pWatsonSharedMemory->wzModuleFileName), wzModuleFileName);
```

Comment:
I see your point.  I think this is fine, but do keep in mind that WS 2008 is a supported platform.  It might be worth opening an issue for the diagnostics team to revisit this in the future.